src/_statusline-semaphore.ts

```
@@ -0,0 +1,180 @@
+	const result = statuslineSemaphoreSchema.safeParse(data);
+	return result.success;
+}
+
+/**
+ * Checks if we should skip execution based on semaphore
+ */
+export function checkShouldSkipExecution(
+	sessionId: string,
+	refreshIntervalMs = STATUSLINE_MIN_REFRESH_INTERVAL_MS,
+): Result.Result<{ shouldSkip: boolean; cachedOutput?: string }, Error> {
+	return Result.try({
```

ESLint

[error] Unsafe call of a(n) `error` type typed value.

(ts/no-unsafe-call)